### PR TITLE
Adjust Docz theme (desktop version)

### PR DESCRIPTION
**Pull request by larimaza**
# What

Adjust the theme of our documentation website according to the design specs.

# Why

We need Astro to look like Astro.

# How

We tried several approaches to a more maintainable/automatic solution, and all of them were too complex:

* Duplicating the Docz default theme to another folder and running that instead: couldn't run the project because of too many problems with all the dependencies and path resolving issues.
* Using Lerna to manage another dependency inside the same repository: also found problems with dependencies and paths.
* Writing a new theme from scratch, following the Docz documentation guide: it works, but would defeat the purpose of using Docz, since we'd have to rewrite the whole thing using React, even the Playground component.

So, after spending several days on these attempts, in order to move faster we decided to use only the `themeConfig` object in `doczrc.js` file, which was created by Docz to customize the current theme without creating a new one.
That's a way less sophisticated solution and it certainly looks hacky, so we'll have to limit theme updates to as little as possible.

This PR only covers desktop version - responsive adjustments will come in another PR. 

# Details

While working on this, I ran Docz on a local server and used developer tools to find the classes that were created for each type of element in the markup - that's the weird class names. That's because I didn't have access to the markup to include my own classes.
Whenever possible, I used element selectors like `h1`.
Most styles only worked if included inside the `body` selector. Please excuse all the hacking needed to override the cascading specificity.
I also added a `h4` with Markdown to the intro text of `.mdx` files, because that was the only way to isolate that part of the content to style it. 
I tried to explain what each weird class does with comments, and hopefully it's readable - but I'm very open to better solutions to organize this file.

This PR also includes a few small adjustments to Astro itself, because while some theme styles were altered, I noticed a few missing styles on the Astro core to make sure components work no matter the context.

**Before** | **After (SO PRETTAY!)**
-|-
Screen Shot 2019-03-25 at 10 58 55|Screen Shot 2019-03-25 at 10 59 06

### CODE REVIEW

**Comment by larimaza** [comment]
[Ajustar o tema do Docz (desktop)](https://trello.com/c/pwbMwAES/67-ajustar-o-tema-do-docz-desktop)

**Comment by larimaza** [comment]
[Ajustar o tema do Docz (desktop)](https://trello.com/c/pwbMwAES/67-ajustar-o-tema-do-docz-desktop)

File: doczrc.js [hunk 1]
```diff
@@ -19,12 +19,148 @@ export default {
...
+        '& .css-wwgiek': { /* Selects menu area */
+          background: 'var(--color-space-100)',
+          borderRight: '1px solid var(--color-space-300)',
+          width: 320,
+          '& .css-pccrlp': { /* Selects inner menu area */
+            width: 320
+          }
+        },
+        '& .css-82rrzo, .css-82rrzo.active, .css-82rrzo:visited': { /* Selects each menu item */
+          fontFamily: "'Poppins', sans-serif",
```
Comment (kuroski) [inline doczrc.js]: Why you didn't use the "font variables" here?

File: doczrc.js [hunk 2]
```diff
@@ -19,12 +19,148 @@ export default {
...
+            display: 'none'
+          }
+        },
+        '& .css-1evit71:after': { /* Removes dashed left border on menu sub-items */
+          display: 'none'
+        }
       },
       h1: {
-        fontFamily: "'Poppins', sans-serif"
+        fontFamily: "'Poppins', sans-serif",
```
Comment (kuroski) [inline doczrc.js]: Why you didn't use the "font variables" here?

File: doczrc.js [hunk 3]
```diff
@@ -19,12 +19,148 @@ export default {
...
+          }
+        },
+        '&#manifesto, &#get-started': { /* Selects specific Homepage h2 headings */
+          fontSize: 40,
+          color: 'var(--color-moon-900)',
+          marginTop: 64
+        },
+      },
+      h4: {
+        fontFamily: "'Lato', sans-serif",
```
Comment (kuroski) [inline doczrc.js]: Why you didn't use the "font variables" here?

File: doczrc.js [hunk 4]
```diff
@@ -19,12 +19,148 @@ export default {
...
+        fontSize: 64,
+        fontWeight: 600,
+        color: 'var(--color-space-100)',
+        margin: '103px 0 186px',
+        ':before': { /* Removes bottom blue line */
+          display: 'none' 
+        }
+      },
+      h2: {
+        fontFamily: "'Poppins', sans-serif",
```
Comment (kuroski) [inline doczrc.js]: Why you didn't use the "font variables" here?

File: doczrc.js [hunk 5]
```diff
@@ -19,12 +19,148 @@ export default {
   themeConfig: {
     styles: {
       body: {
-        fontFamily: "'Poppins', sans-serif",
-        fontSize: 16,
-        lineHeight: 1.5
+        '& .css-pw2o4y': { /* Selects page wrapper div for gradient header */
```
Comment (kuroski) [inline doczrc.js]: I think I have a suggestion about how we can give this selectors a meaning.
Maybe you can abstract that to a different file or even just create a constant and work like this:

``` javascript 
const selectors = {
  wrapper: '& .css-pw2o4y',
  logo: '& .css-c9p757'
}

export default {
  // all docz configs ......
  themeConfig: {
    styles: {
      body: {
        [selectors.wrapper]: {....},
        [selectors.logo]: {
          height: 171,
          width: 151,
          margin: '46px 60px',
          backgroundImage: 'url("public/astro-logo.svg")',
          backgroundSize: 200,
          backgroundPosition: 'center',
          backgroundRepeat: 'no-repeat',
          textIndent: -999
        }
      }
    }
  }
}
```

what do you think about that?

File: doczrc.js (hunk 4 above)
Comment (larimaza) [reply doczrc.js]: For some reason, in this context font variables only work if you're going to use the whole variable without adding any more separate settings. :/ if you add extra, some properties don't work!

File: doczrc.js (hunk 3 above)
Comment (larimaza) [reply doczrc.js]: For some reason, in this context font variables only work if you're going to use the whole variable without adding any more separate settings like changing the font size. :/ if you add extra, some properties don't work!

File: doczrc.js (hunk 2 above)
Comment (larimaza) [reply doczrc.js]: For some reason, in this context font variables only work if you're going to use the whole variable without adding any more separate settings like changing the font size. :/ if you add extra, some properties don't work!

File: doczrc.js (hunk 1 above)
Comment (larimaza) [reply doczrc.js]: For some reason, in this context font variables only work if you're going to use the whole variable without adding any more separate settings like changing the font size. :/ if you add extra, some properties don't work!

File: doczrc.js (hunk 5 above)
Comment (larimaza) [reply doczrc.js]: Definitely looks more readable and still works! Great idea, done on a4ac2e9. I've extracted almost everything, except for stuff that looked better if left in place. What do you think?

File: public/astro-logo.svg [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+                    <polygon id="Fill-348" fill="#74E24A" points="34.0852349 63.8 44.8489933 63.8 44.8489933 53.1666667 34.0852349 53.1666667"></polygon>
+                    <polygon id="Fill-348" fill="#E078D6" points="186.571812 10.6333333 197.33557 10.6333333 197.33557 0 186.571812 0"></polygon>
+                </g>
+            </g>
+            <text id="Magnetis’-Design-Sys" font-family="Poppins-Regular, Poppins" font-size="24" font-weight="normal" line-spacing="18" fill="#597183">
```
[the remark this answers is not in the document]
Comment (larimaza) [reply public/astro-logo.svg]: Fixed on 1e86fb8, see if it works now!

File: public/drop-up.svg [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (larimaza) [reply public/drop-up.svg]: 1e86fb8

File: public/drop-down.svg [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (larimaza) [reply public/drop-down.svg]: 1e86fb8

File: public/astro-logo.svg [hunk 9]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (larimaza) [reply public/astro-logo.svg]: Great point - Done on 1e86fb8

[the remark this answers is not in the document]
Comment (larimaza) [reply public/astro-logo.svg]: Also LOVE the tool you linked here!

**Pull request merged**